Deprecate SlavePort and MasterPort Classes and change terminology

shparekh00 opened the pull request:
Changing master slave terminology.

https://gem5.atlassian.net/browse/GEM5-9

Change-Id: I5e6e90b2916df4dbfccdaabe97423f377a1f6e3f

shparekh00 committed:
mem: Deprecated slaveport and masterport classes

Change-Id: I5e6e90b2916df4dbfccdaabe97423f377a1f6e3f

ebrickey commented on src/mem/port.hh:
```diff
@@ -373,52 +383,52 @@ class SlavePort : public Port, public AtomicResponseProtocol,
+     * Called by the requestor port to unbind. Should never be called
      * directly.
      */
     void slaveUnbind();
```
There were two functions (slaveUnbind() and slaveBind()) we were unsure how to deprecate. In the RequestPort class should this now be RequestPort::requestorUnbind() and in the SlavePort class we'd need SlavePort::slaveUnbind(). Both of which would need to be identical functions with the same contents just a different name that we need to define in the .cc? These were the only two cases we've seen so far where the old terminology effected function names in this class.

powerjg replied:
You should be able to just change the names of these functions. This shouldn't be an exposed API, and we should be able to change it however we feel.

ebrickey replied:
Wouldn't some references to the oldAPI we've yet to revise still call SlavePort::slaveUnbind()?

ebrickey replied:
Actually looking at it closer, it looks like slaveUnbind() is only called in token_port.cc and slaveBind() is never called. In that case I think we could just change the name as you suggested and update the only reference. The description on the function says  "Called by the requestor port to unbind. Should never be called directly." which seems odd since we only found one reference and for only one of the functions?

ebrickey committed:
misc: learning_gem5 simple_memobj refactor

Change-Id: I2fe5eada5ca75d54e6989b9c97c0e0cbbf73d94c

ebrickey committed:
misc: Updated port classes & refs to remove slaveBind()/UnBind()

Change-Id: I9106397b8816d8148dd916510bbcf65ed499d303

ebrickey committed:
misc: comment fix in simple_memobj

Change-Id: I3a58b6f226d6fe74c8b7976acb95af1aa665dd9e

powerjg commented on src/mem/port.cc:
```diff
@@ -103,54 +103,55 @@ class DefaultSlavePort : public SlavePort
     AddrRangeList getAddrRanges() const override { return AddrRangeList(); }
 };
 
-DefaultMasterPort defaultMasterPort;
```
Are these variable names used anywhere in gem5?

ebrickey replied:
I searched the gem5 directory for other references and it appears to only be used in port.cc. During our discussion yesterday you renamed a variable and were able to see/rename all other references in the repo. I thought the option was "rename symbol" however when I select this option vscode never successfully finds the references it just has this loading message.
![image](https://user-images.githubusercontent.com/35640953/88855740-2beec280-d1a8-11ea-83b5-32433e17a6ea.png)

powerjg commented on src/mem/port.hh:
```diff
@@ -68,37 +71,37 @@ class SlavePort;
 
-  private:
-    SlavePort *_slavePort;
+    private:
```
Added spaces here. I know it's weird and IDEs complain, but our style says that private/public/protected are *two* spaces indented.

ebrickey replied:
Fixed!

powerjg commented on src/mem/port.hh:
```diff
@@ -242,56 +245,63 @@ class MasterPort : public Port, public AtomicRequestProtocol,
 
+class M5_DEPRECATED MasterPort : public RequestPort
+{
+    public:
```
two spaces :)

ebrickey replied:
Fixed! Let me know if that's not quite what you had in mind.

powerjg reviewed (commented):
LGTM! This is perfect so far (assuming things still work ;)). I have a couple of minor comments below.

You may want to get in the habit of running a subset of the regression tests, too. Check out TESTING.md for details (or ask Bobby).

ebrickey committed:
misc: spacing fixes

Change-Id: I93586363d252851832ce76d3c29fd22709576bab

shparekh00 committed:
misc: Replaced MasterPort and SlavePort in src/arch

Change-Id: I06595a97eea050d2a025b4331738a618410b7779

shparekh00 committed:
misc: Replaced MasterPort and SlavePort in src/arch

Doesn't compile because it depends on changes in src/dev
Change-Id: I06595a97eea050d2a025b4331738a618410b7779

shparekh00 committed:
misc: Replaced MasterPort and SlavePort in src/arch

Doesn't compile because it depends on changes in src/dev
Change-Id: I06595a97eea050d2a025b4331738a618410b7779

powerjg commented on src/mem/port.hh:
```diff
@@ -438,71 +448,79 @@ class SlavePort : public Port, public AtomicResponseProtocol,
 
+class M5_DEPRECATED SlavePort : public ResponsePort
+{
+    public:
```
Two spaces instead of four
```suggestion
  public:
```

ebrickey replied:
Fixed, thank you for catching this!

powerjg commented on src/mem/port.hh:
```diff
@@ -438,71 +448,79 @@ class SlavePort : public Port, public AtomicResponseProtocol,
+class M5_DEPRECATED SlavePort : public ResponsePort
+{
+    public:
+        SlavePort(const std::string& name, SimObject* _owner,
```
Four spaces, not eight :)
```suggestion
    SlavePort(const std::string& name, SimObject* _owner,
```

ebrickey replied:
Got it! How do we indent lines where we're past the character limit for a line? For example, like the line below this comment.

powerjg replied:
Very carefully ;). 

I suggest looking within in the file or at other files in the directory and copying that style.

Generally, I would align the next line with the parentheses. The main exception to this would be for `if` statements where you can't tell what is the body and what is the continuation of the condition. Another time this doesn't work is if the parenthesis starts at the end of the line (e.g., >60 chars). 

It's a judgement call... Basically, try to do something that looks the least bad :). The goal is readability. Remember, code is going to be read 10x more than it's written.

powerjg reviewed (commented):
one tiny thing below, but otherwise this is perfect!

Commit:
misc: Changed 'master/slave' terminology

Changed local variables or references in files in cpu that used 'master/slave' -> 'requestor/responder'

Change-Id: Ic79735355efc0d9a8c8a2be603478cb3bb3c31e0

Commit:
Revert "misc: Changed 'master/slave' terminology"

This reverts commit abe6e798545f3e09af7aa56f7cd2cbf2469f343b.

shparekh00 committed:
misc: Merge branch, change VectorResponsePort to VectorSlavePort

'shivani/REU/reqresp' https://github.com/darchr/gem5 into shivani/REU/reqresp

Change-Id: I9042553324c5a7bd969c47f9fc4483188e1dcda8

shparekh00 committed:
misc: Merge branch, change VectorResponsePort to VectorSlavePort

'shivani/REU/reqresp' https://github.com/darchr/gem5 into shivani/REU/reqresp

Change-Id: I9042553324c5a7bd969c47f9fc4483188e1dcda8

shparekh00 committed:
misc: Merge branch

'shivani/REU/reqresp' https://github.com/darchr/gem5 into shivani/REU/reqresp

Change-Id: Ia7012aadcefe84a6adeb0f4c4d3f2f68ccece9d0

ebrickey committed:
misc: spacing style fixes in port

Change-Id: I3447717fe65be20f54bacdc9e106ff4bcba37310

shparekh00 committed:
misc: Modified Master/SlavePort in src/mem

snoop_filter.cc, snoop_filter.hh, token_port.cc, token_port.hh

Change-Id: Iec27c02b7cf01a70937b60605386f43971eaafdf

shparekh00 committed:
misc: Merge branch 'shivani/REU/reqresp'

Change-Id: I98f8c0dc73669775e36d1042b2038875eccb7139

shparekh00 committed:
misc: Merge shivani/REU/reqresp

Change-Id: Ie593524e756df996f4ffc0d1709bc6804d9837eb

shparekh00 committed:
misc: Merge shivani/REU/reqresp

Change-Id: Ie593524e756df996f4ffc0d1709bc6804d9837eb

ebrickey committed:
misc: merged + updated to use responderUnbind()

Change-Id: Ie8c7574fe8dc8b1ac4e396e9f11a880ba97390cb

shparekh00 committed:
misc: Merge branch 'REU/reqresp' into origin/REU/reqresp

Change-Id: Icbd34d21c1f4119746b83bbb5dac580f8aa689fb

Commit:
misc: Merge sub branch to REU/reqresp

Made changes to Master/SlavePort terminology and merge sub branch

Change-Id: Icb60101c3f1a5c60956316b59c7c64f6b903fbcc

shparekh00 committed:
misc: Merge local REU/reqresp to origin

Change-Id: I951acd84a97258f080d8e15edbd78a9a0bc02a76
